cfgmgr/vlanmgrd.cpp

```diff
@@ -0,0 +1,107 @@
+#include <unistd.h>
+#include <vector>
+#include <sstream>
+#include <fstream>
+#include <iostream>
+#include <mutex>
+#include "dbconnector.h"
+#include "select.h"
+#include "exec.h"
+#include "schema.h"
+#include "macaddress.h"
+#include "producerstatetable.h"
+#include "vlanmgr.h"
+#include "shellcmd.h"
+
+using namespace std;
+using namespace swss;
+
+/* select() function timeout retry time, in millisecond */
+#define SELECT_TIMEOUT 1000
+
+MacAddress gMacAddress;
+
+/*
+ * Following global variables are defined here for the purpose of
+ * using existing Orch class which is to be refactored soon to
+ * eliminate the direct exposure of the global variables.
+ *
+ * Once Orch class refactoring is done, these global variables
+ * should be removed from here.
+ */
+int gBatchSize = 0;
+bool gSwssRecord = false;
+bool gLogRotate = false;
+ofstream gRecordOfs;
+string gRecordFile;
+/* Global database mutex */
+mutex gDbMutex;
+
+int main(int argc, char **argv)
+{
+    Logger::linkToDbNative("vlanmgrd");
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_NOTICE("--- Starting vlanmgrd ---");
+
+    string mac_str;
+    stringstream cmd;
+    cmd << IP_CMD << " link show eth0 | " << GREP_CMD << " ether | " << AWK_CMD << " '{print $2}'";
```

Comment:
we need to use the -m option to get the mac address since it is complicated. 

You can check here. [https://github.com/Azure/sonic-buildimage/blob/37dc7bd4786c793796530e53da304b5e68b7ed16/files/image_config/interfaces/interfaces-config.sh#L7](https://github.com/Azure/sonic-buildimage/blob/37dc7bd4786c793796530e53da304b5e68b7ed16/files/image_config/interfaces/interfaces-config.sh#L7) 

For some asic, the switch mac must match certain pattern. 
Here for the VLAN mac, you will want it to be the same as the switch mac. Therefore, we need to follow the same procedure.

Reply:
Ok, I'm think of using the switch table in configDB with has switch mac field

127.0.0.1:6379[4]> hgetall "SWITCH|SWITCH_ATTR"
1) "switch_mac"
2) "00:05:64:30:73:c0"

In interfaces-config.sh,  we set the switch_mac to configDB if it is not there already, then use the configDB data in vlanMgr.   This way it give people option to specify switch mac other than that of eth0.

Reply:
that is fine, but use eth0 mac is not an option. need to remove that part.

Reply:
done.